lib/facter/resolvers/windows/networking.rb

```diff
@@ -130,6 +132,14 @@ def find_primary_interface(sock_addr, name, addr)
               @fact_list[:primary_interface] = name.to_s
             end
           end
+
+          def retrieve_domain_from_registry
+            reg = ::Win32::Registry::HKEY_LOCAL_MACHINE.open('SYSTEM\\CurrentControlSet\\Services\\Tcpip\\Parameters')
+            reg.each do |name, _value|
+              @fact_list[:domain] = reg[name] if name == 'Domain'
+            end
+            reg.close
```

Comment:
You can use a block to avoid explicitly closing the register:
```ruby
::Win32::Registry::HKEY_LOCAL_MACHINE.open('SYSTEM\\CurrentControlSet\\Services\\Tcpip\\Parameters') do |reg|
  ...
end
```

eg: ::Win32::Registry::HKEY_LOCAL_MACHINE.open('SYSTEM\\CurrentControlSet\\Services\\Tcpip\\Parameters')

Should this handle registry read errors, like `Win32::Registry::Error` ?

Reply:
Done